apps/site/lib/site_web/controllers/stop_controller.ex

```diff
@@ -192,19 +192,44 @@ defmodule SiteWeb.StopController do
     json(conn, route_patterns_by_route_and_headsign(stop_id))
   end
 
-  @type by_route_and_headsign :: %{Route.id_t() => %{String.t() => [RoutePattern.t()]}}
+  @type headsign_info :: %{
+          required(:direction_id) => 0 | 1,
+          required(:route_patterns) => [RoutePattern.t()]
+        }
+  @type by_route_and_headsign :: %{Route.id_t() => %{String.t() => headsign_info}}
```

Comment:
we don't do separately versioned backends and frontends here, they'll be deployed together 😉

Reply:
I may be missing a nuance here, but my previous understanding on Skate was that this would be considered a breaking change. Even though the frontend and backends are deployed together, there will be a period of time where some ecs instances have the old backend and some instances have the new backend, during which someone using the stop page has no control over whether the new or old instances are hit, so there could be a mismatch between the data that is returned and what the frontend expects. Is that situation avoided with how dotcom is set up?

Reply:
we've never thought to mitigate this before! but when you explain it here it makes sense...aah 🫠 🫠 🫠 🫠

Reply:
I do think that problem might be less frequently encountered on Dotcom, because folks don't have it open very long per session (just a few minutes -- I presume folks working in Skate maintain a much longer working session during typical use). I'll have to figure out how to handle that more systematically but for the sake of this PR will consider it out of scope. But thanks so much for raising this point!!